src/lib/indexFolderToObjects.ts

```diff
@@ -0,0 +1,79 @@
+import { getUniqueValues, recursiveWalkDir } from "../utils/index.js";
+import type { WikiLink } from "../utils/index.js";
+import { extractFileSchemeFromObject } from "../utils/extractFileSchemeFromObject.js";
+import { readLocalMarkdownFileToObject } from "./readLocalMarkdownFileToObject.js";
+import type { File, FileTag, Link } from "./types/schemaTypes.js";
+
+export function indexFolderToObjects(
+  folderPath: string,
+  pathToUrlResolver: (filePath: string) => string,
+  ignorePatterns?: RegExp[]
+) {
+  const filePathsToIndex = recursiveWalkDir(folderPath);
+  const files: File[] = [];
+  const tags: string[] = [];
+  const fileTags: FileTag[] = [];
+  const links: Link[] = [];
+  const filteredFilePathsToIndex = filePathsToIndex.filter((filePath) =>
+    shouldIncludeFile(filePath, ignorePatterns)
+  );
+
+  for (const filePath of filteredFilePathsToIndex) {
+    const fileObject = readLocalMarkdownFileToObject(
+      folderPath,
+      filePath,
+      filePathsToIndex,
+      pathToUrlResolver
+    );
```

Comment:
This looks a bit awkward imo... Both the naming, the arguments it takes and unnecesarily adds a layer of abstraction. I wonder if we could get rid of the `readLocalMarkdownFileToObject` whatsoever. This could just be:

```suggestion
    const id = generateFileIdFromPath(filePath);
    const extension = getFileExtensionFromPath(filePath);
    
    if (MddbFile.supportedExtensions.includes(extension)) {
      return { id, extension, filePath };
   }
   
   const pathRelativeToFolder = path.relative(folderPath, filePath);
   const urlPath = pathToUrlResolver(pathRelativeToFolder);
   
   const data = parseMarkdownFile(filePath)
   ...
```

Reply:
Yes, I agree.
This could benefit from a slight refactor...